projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3772,8 +3768,8 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
         }
         this.resetColumnsCaches();
         this.resetColumnCollections();
-        this.resetCachedWidths();
         this.resetForOfCache();
+        this.resetCachedWidths();
```

Comment:
Didn't see a test that specifically relates to this, wondering if it's relevant

Reply:
It's related to tests:
 * `should correctly initialize when there are initially pinned columns` 
 * `should correctly initialize when there are initially pinned columns` 

It is due to the `resetCachedWidths()` using the grid API in order to calculate the grid container size and when the test scenario calls `grid.unpinnedWidth` the last width calculated doesn't include if scrollbar is rendered, since the widths are not reset after `resetCachedWidths()`.

The initial issue was that the grid throws error when columns don't have width set and was addressed with 0c09e9f commit.